File: crates/tau-sdk-codegen/tests/fixtures/tool_agent/project.ts
import { agent, models, tool } from "tau";

export const m = models({ haiku: { backend: "anthropic", model: "claude-haiku-4-5" } });

const read_temp = tool({
    native: "ReadTemp",
    description: "Read the temperature sensor",
    capabilities: [{ kind: "fs.read", paths: ["/sys/**"] }],
});

export const monitor = agent({
    display_name: "Monitor",
    package: "monitor@^0.1",
    model: "haiku",
    prompt: { system: "Watch the temperature; report if above 30C." },
    tools: { read_temp },
});

File: crates/tau-sdk-codegen/tests/fixtures/tool_agent/project.py
from tau_sdk import agent, models, model, tool, print_toml

m = models(haiku=model(backend="anthropic", model="claude-haiku-4-5"))

read_temp = tool(
    native="ReadTemp",
    description="Read the temperature sensor",
    capabilities=[{"kind": "fs.read", "paths": ["/sys/**"]}],
)

monitor = agent(
    display_name="Monitor",
    package="monitor@^0.1",
    model="haiku",
    prompt={"system": "Watch the temperature; report if above 30C."},
    tool_refs=["read_temp"],
)

print_toml(
    project="tool-agent",
    models=m,
    agents={"monitor": monitor},
    tools={"read_temp": read_temp},
)
